src/main/java/mate/academy/spring/controller/OrderController.java

```diff
@@ -0,0 +1,50 @@
+package mate.academy.spring.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.mapper.DtoResponseMapper;
+import mate.academy.spring.mapper.impl.response.OrderResponseMapper;
+import mate.academy.spring.model.Order;
+import mate.academy.spring.model.dto.response.OrderResponseDto;
+import mate.academy.spring.service.OrderService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final DtoResponseMapper<OrderResponseDto, Order> orderResponseMapper;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+
+    public OrderController(OrderService orderService,
+                           OrderResponseMapper orderResponseMapper,
```

Comment:
switch to abstraction
DtoResponseMapper<OrderResponseDto, Order> orderResponseMapper